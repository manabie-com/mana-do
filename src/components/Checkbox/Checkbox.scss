@import "../../assets/styles/main.scss";

.success-checkmark {
  width: 5rem;
  height: 5rem;
  margin: 0 auto;
  cursor: pointer;
  transform: scale(0.32, 0.32);

  .check-icon {
    width: 5rem;
    height: 5rem;
    position: relative;
    border-radius: 50%;
    box-sizing: content-box;
    border: 0.25rem solid $green;

    &::before {
      top: 3px;
      left: -2px;
      width: 30px;
      transform-origin: 100% 50%;
      border-radius: 100px 0 0 100px;
    }

    &::after {
      top: 0;
      left: 30px;
      width: 60px;
      transform-origin: 0 50%;
      border-radius: 0 100px 100px 0;
      animation: rotate-circle 4.25s ease-in;
    }

    &::before,
    &::after {
      content: "";
      height: 100px;
      position: absolute;
      background: #ffffff;
      transform: rotate(-45deg);
    }

    .icon-line {
      height: 5px;
      background-color: $green;
      display: block;
      border-radius: 0.125rem;
      position: absolute;
      z-index: 10;

      &.line-tip {
        top: 46px;
        left: 14px;
        width: 25px;
        transform: rotate(45deg);
        animation: icon-line-tip 0.75s;
      }

      &.line-long {
        top: 38px;
        right: 0.5rem;
        width: 47px;
        transform: rotate(-45deg);
        animation: icon-line-long 0.75s;
      }
    }

    .icon-circle {
      top: -4px;
      left: -4px;
      z-index: 10;
      width: 5rem;
      height: 5rem;
      border-radius: 50%;
      position: absolute;
      box-sizing: content-box;
      border: 0.25rem solid rgba(76, 175, 80, 0.5);
    }

    .icon-fix {
      top: 0.5rem;
      width: 5px;
      left: 26px;
      z-index: 1;
      height: 85px;
      position: absolute;
      transform: rotate(-45deg);
      background-color: #ffffff;
    }
  }
}
.icon-circle {
  z-index: 10;
  width: 5rem;
  height: 5rem;
  border-radius: 50%;
  position: relative;
  box-sizing: content-box;
  border: 0.25rem solid rgba(76, 175, 80, 0.5);
}

@keyframes rotate-circle {
  0% {
    transform: rotate(-45deg);
  }
  5% {
    transform: rotate(-45deg);
  }
  12% {
    transform: rotate(-405deg);
  }
  100% {
    transform: rotate(-405deg);
  }
}

@keyframes icon-line-tip {
  0% {
    width: 0;
    left: 1px;
    top: 19px;
  }
  54% {
    width: 0;
    left: 1px;
    top: 19px;
  }
  70% {
    width: 50px;
    left: -0.5rem;
    top: 37px;
  }
  84% {
    width: 17px;
    left: 21px;
    top: 3rem;
  }
  100% {
    width: 25px;
    left: 14px;
    top: 45px;
  }
}

@keyframes icon-line-long {
  0% {
    width: 0;
    right: 46px;
    top: 3.375rem;
  }
  65% {
    width: 0;
    right: 46px;
    top: 3.375rem;
  }
  84% {
    width: 55px;
    right: 0px;
    top: 35px;
  }
  100% {
    width: 47px;
    right: 0.5rem;
    top: 38px;
  }
}
