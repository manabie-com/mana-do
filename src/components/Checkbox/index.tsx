import React from 'react';

interface checkboxProps {
    onChange: (e: any, todoId: string) => void;
    className?: string | undefined;
    checked?: boolean;
    todoId: string;
}

const Checkbox = (props: checkboxProps) => {
    const { className, onChange, checked, todoId } = props

    return (
        <input
            checked={checked}
            className={`${className} `}
            onChange={(e) => onChange(e, todoId)}
            type="checkbox"

        />
    )

}

export default Checkbox