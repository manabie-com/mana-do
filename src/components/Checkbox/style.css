/* CUSTOM CHECKBOX */
.label {
  display: flex;
  align-items: center;
  cursor: pointer;
  position: relative;
}

.label > span {
  color: #424242;
  font-weight: 700;
  cursor: pointer;
  padding: 0.5rem 0.25rem;
  word-break: break-all;
}

.label > input[type="checkbox"] {
  height: 18px;
  width: 18px;
  -webkit-appearance: none;
  -moz-appearance: none;
  -o-appearance: none;
  appearance: none;
  border: 1px solid #424242;
  border-radius: 4px;
  outline: none;
  transition-duration: 0.3s;
  background-color: #fff;
  cursor: pointer;
}

.label > input:checked {
  border: 1px solid lightcoral;
  background-color: lightcoral;
}

.label > input:checked + span::before {
  content: '\2713';
  display: block;
  text-align: center;
  color: #fff;
  position: absolute;
  left: 8px;
  top: 8px;
}

.content {
  flex: 1 1;
  text-align: left;
  margin-left: 8px;
  cursor: pointer;
  letter-spacing: 2px;
  font-weight: 500;
}

.content-checked{
  text-decoration: line-through;
}