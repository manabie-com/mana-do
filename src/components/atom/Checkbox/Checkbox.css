.Checkbox__item {
  display: flex;
  align-items: center;
  cursor: pointer;
}

.Checkbox__label {
  flex: 1 1;
  text-align: left;
  margin-left: 8px;
  font-size: 14px;
}
