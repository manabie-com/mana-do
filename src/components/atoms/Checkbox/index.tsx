import React from 'react';

interface CheckboxProps extends React.HTMLAttributes<any>{
    className?:string
}

export const Checkbox: React.FC<CheckboxProps> =(props)=>{
    return (
        <input {...props} type="checkbox" />
    )
}