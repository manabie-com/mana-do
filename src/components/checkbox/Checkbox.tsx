import { InputType } from "models";
import React, { FC, memo } from "react";
import "./styles.scss";

export interface ICheckboxProps {
  checked?: boolean;
  id?: string;
  className?: string;
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const Checkbox: FC<ICheckboxProps> = ({
  checked,
  onChange,
  className,
  id,
  ...props
}) => {
  return (
    <input
      type={InputType.CHECKBOX}
      checked={checked}
      onChange={onChange}
      data-testid="checkbox"
      id={id || ""}
      className={`cbx-todo ${className ? className : ""}`}
      {...props}
    />
  );
};

export default memo(Checkbox);
