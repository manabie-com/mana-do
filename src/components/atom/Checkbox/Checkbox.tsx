import React from "react";
import "./Checkbox.css";

export interface CheckboxProps extends React.HTMLAttributes<HTMLInputElement> {
  checked: boolean;
  onChange: React.ChangeEventHandler<HTMLInputElement>;
  label?: string;
  name: string;
  id: string;
  fullWidth?: boolean;
}

const Checkbox: React.FC<CheckboxProps> = ({
  checked,
  onChange,
  label,
  name,
  id,
  fullWidth = false,
}) => {
  return (
    <React.Fragment>
      <label
        htmlFor={id}
        className={`Checkbox__item ${fullWidth ? "w-100" : ""}`}
      >
        <input
          name={name}
          id={id}
          type="checkbox"
          checked={checked}
          onChange={onChange}
        />
        {label && <span className="Checkbox__label">{label}</span>}
      </label>
    </React.Fragment>
  );
};

export default Checkbox;
