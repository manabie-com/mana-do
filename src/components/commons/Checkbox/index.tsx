import React, { forwardRef, useRef } from "react";
import cn from "classnames";
import CheckboxChecked from "./icon/CheckboxChecked";
import CheckboxUnChecked from "./icon/CheckboxUnChecked";
import Input from "../Input";
import "./style.css";

interface Props {
  className?: string;
  iconRef?: React.Ref<unknown>;
  disabled?: boolean;
  children?: React.ReactNode;
  checked?: boolean;
  value?: string;
  name?: string;
  onChange?: (e: React.ChangeEvent<HTMLInputElement>) => void;
}

const defaultProps: Props = {
  disabled: false,
  checked: false,
};

const Checkbox = forwardRef((props: Props, ref: any) => {
  const {
    className,
    iconRef,
    name,
    disabled,
    checked,
    value,
    children,
    onChange,
    ...rest
  } = {
    ...defaultProps,
    ...props,
  };

  const classOfRoot = cn(
    "checkbox-custom",
    className,
    disabled && "checkbox-custom-disabled",
    checked && "checkbox-custom-checked"
  );

  const refOfInput = useRef();

  return (
    <label
      data-testid="checkbox-custom"
      className={classOfRoot}
      ref={ref}
      {...rest}
    >
      {checked ? (
        <CheckboxChecked ref={iconRef} />
      ) : (
        <CheckboxUnChecked ref={iconRef} />
      )}

      {children}
      <Input
        readOnly
        type="checkbox"
        ref={refOfInput}
        name={name}
        value={value}
        className="checkbox-custom-input"
        checked={checked}
        onChange={onChange}
      />
    </label>
  );
});

export default Checkbox;
