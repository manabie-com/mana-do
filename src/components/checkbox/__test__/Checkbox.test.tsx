import { render, RenderResult } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import React from "react";
import { getElementByTestId } from "utils/testUtils";
import Checkbox, { ICheckboxProps } from "../Checkbox";

const onChangeAction = jest.fn();

const renderComponent = ({ ...props }: ICheckboxProps): RenderResult => {
  return render(<Checkbox {...props} />);
};

describe("<Checkbox /> rendering", () => {
  it("should match snapshot", () => {
    const wrapper = renderComponent({
      onChange: onChangeAction,
    });
    expect(wrapper.container).toMatchSnapshot();
  });

  it("should render correcly", () => {
    renderComponent({
      onChange: onChangeAction,
    });
    const element = getElementByTestId("checkbox");
    expect(element).toBeInTheDocument();
  });

  it("should render a checkbox with the default class", () => {
    renderComponent({
      onChange: onChangeAction,
    });
    const element = getElementByTestId("checkbox");
    expect(element).toHaveClass("cbx-todo");
  });
});

describe("<Checkbox /> interacting", () => {
  it("should render a checkbox clickable", () => {
    renderComponent({
      onChange: onChangeAction,
    });
    const element = getElementByTestId("checkbox");
    expect(element).not.toBeDisabled();
  });

  it("should handle onChange when clicked", () => {
    renderComponent({
      onChange: onChangeAction,
    });
    const element = getElementByTestId("checkbox");
    userEvent.click(element);
    expect(onChangeAction).toHaveBeenCalled();
  });
});
