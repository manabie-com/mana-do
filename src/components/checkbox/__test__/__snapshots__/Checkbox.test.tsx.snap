// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> rendering should match snapshot 1`] = `
<div>
  <input
    class="cbx-todo "
    data-testid="checkbox"
    id=""
    type="checkbox"
  />
</div>
`;
